src/playback-watcher.js

```diff
@@ -115,6 +121,31 @@ export default class PlaybackWatcher {
   }
 
   /**
+   * Fixes situations where there's a bad seek
+   *
+   * @return {Boolean} whether an action was taken to fix the seek
+   * @private
+   */
+  fixBadSeeks_() {
+    let seekable = this.seekable();
+    let currentTime = this.tech_.currentTime();
+
+    if (this.tech_.seeking() &&
```

Comment:
Should this function ever run in VOD?

It seems wise to check to see if the duration is "Infinity" before taking any other measures.